src/snowflake/cli/plugins/nativeapp/codegen/snowpark/extension_function_utils.py

```diff
@@ -5,14 +5,253 @@
     List,
     Optional,
     Tuple,
+    Type,
     Union,
 )
 
-from snowflake.cli.api.console import cli_console as cc
+from click.exceptions import ClickException
 
+
+class MalformedExtensionFunctionError(ClickException):
+    """Required extension function attribute is missing."""
+
+    def __init__(self, message: str):
+        super().__init__(message=message)
+
+
+# This prefix is created by Snowpark for an extension function when the user has not supplied any themselves.
+# TODO: move to sandbox execution to omit object name in this case: https://github.com/snowflakedb/snowflake-cli/pull/1056/files#r1599784063
 TEMP_OBJECT_NAME_PREFIX = "SNOWPARK_TEMP_"
 
 
+def get_object_type_as_text(name: str) -> str:
+    """
+    Replace underscores with spaces in a given string.
+
+    Parameters:
+        name (str): Any arbitrary string
+    Returns:
+        A string that has replaced underscores with spaces.
+    """
+    return name.replace("_", " ")
+
+
+def _sanitize_ex_fn_str_attribute(
```

Comment:
nit: I would either remove `ex_fn` here (since the module name makes it clear that extension functions are the scope), or expand it. I don't think this name will be clear in the long run, after we're done with the initial feature.